File: packages/protocol-parser/src/hexToTableSchema.ts
import { Hex, sliceHex } from "viem";
import { TableSchema } from "./common";
import { hexToSchema } from "./hexToSchema";

export function hexToTableSchema(data: Hex): TableSchema {
  const valueSchema = hexToSchema(sliceHex(data, 0, 32));
  const keySchema = hexToSchema(sliceHex(data, 32, 64));
  return {
    keySchema,
    valueSchema,
  };
}

File: packages/services/pkg/protocol-parser/hexToTableSchema.go
package protocolparser

func HexToTableSchema(data string) TableSchema {
	valueSchema := HexToSchema(HexSlice(data, 0, 32))
	keySchema := HexToSchema(HexSlice(data, 32, 64))

	return TableSchema{
		KeySchema:   keySchema,
		ValueSchema: valueSchema,
	}
}
